src/core/layertree/qgslayertreelayer.h

```diff
@@ -128,6 +128,20 @@ class CORE_EXPORT QgsLayerTreeLayer : public QgsLayerTreeNode
      */
     void resolveReferences( const QgsProject *project, bool looseMatching = false ) override;
 
+    /**
+     * set the expression to evaluate
+     *
+     * \since QGIS 3.8
+     */
+    void setExpression( const QString &expression );
```

Comment:
What do you think about this change to make it more explicit?

```suggestion
    void setLabelExpression( const QString &expression );
```

Reply:
done